docs/content/connector/connector_protocol.md

```diff
@@ -0,0 +1,657 @@
+      "content": "Any content"
+   }
+}
+```
+
+The content type of **furtherStep** can have one of the values described [below](#instantiation-step-types).
+
+If the connector responds with status code **201** the payload is ignored, because the status code indicates a successful instantiation.
+If further steps are required, the connector should instead respond with status code **202**.
+
+### Instance removal
+
+```http
+DELTETE /-instanceCallbackURL-/-instanceID- HTTP/1.1
+Signature: -base64encodedSignature-
+```
+
+Whenever an instance is removed on the connctd platform, a DELETE request is send to the connector.
+For more information on the signature read the documentation for [new installations](#new-installation).
+
+#### Response
+
+The connector should take appropriate actions on his side, e.g. remove the instance, and respond with status code **204** (**NO_CONTENT**).
+If another status code is returned, an error message is returned to the app that is removing the instance, but the removal will still succeed.
+
+### Instantiation States
+
+* 1: INITIALIZED
+
+   Set by connctd when the instantiation of a connector was triggered.
+
+* 2: COMPLETE
+
+   Set by connctd as soon as the external connector service replied with 201 (Created) to the instantiation request.
+   A connector is instantiated correctly only if it is in this state.
+
+* 3: ONGOING
+
+   Set by connctd as soon as the external connector service replied with 202 (Accepted) to the instantiation request.
+   This indicates that external connector requires additional input.
+   Instructions on further steps can be expressed by providing details and a further step field in the response to connctd.
+
+* 4: FAILED
+
+   Set by connctd as soon as the external connector service replied with 403 (Forbidden) or 401 (Unauthorized) to the instantiation request.
+   This gives the external connector the possibility to signalize faulty configuration parameters (e.g. entered by developer).
+
+* 5: ERROR
+
+   Set by connctd as soon as the external connector service replied with an unexpected status code or when the response is malformed.
+   In this case connctd responds with its own state details which should give external connector developers information on how to fix the problems.
+
+* 6: INTERNAL_ERROR
+
+   Set by connctd due to an internal issue.
+   Leads to the cancellation of the instantiation.
+
+* 7: REMOVAL_ONGOING
+
+   Set by connctd as soon as an instantiation was marked for removal.
+
+#### Instantiation step types
+
+* 1: TEXT
+
+   The corresponding value is interpreted as text.
+
+* 2: MARKDOWN
+
+   The corresponding value is interpreted as markdown.
+
+* 3: REDIRECT
+
+   The corresponding value is expected to be an URL.
+
+### Action callback
+
+Whenever an action is triggered on a thing, the connector is informed in order to implement the action.
+The following request is sent against the ***actionCallbackUrl*** if an action is triggered.
```

Comment:
```suggestion
The following request is sent against the ***actionCallbackUrl*** if an action is triggered. Every action request has a unique action id. This allows a connector to implement even long-taking actions since the connector can always reference a specific action request and finalize it whenever the actual action was executed.
```